x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/exchange/ExchangeService.java

```diff
@@ -277,9 +278,16 @@ protected void doRun() {
      * @param exchangeId       the exchange ID
      * @param transportService the transport service
      * @param conn             the connection to the remote node where the remote exchange sink is located
+     * @param failureCollector if not null, the failures will be sent to this consumer and the sink will be marked as finished
```

Comment:
Then the same happens as before - the failure is kicked up the listener chain and somewhere it will hit the handler which will abort the whole thing and clean everything up. There's a number of them around the code, but which one it'll hit depends on when the failure happens IIRC.